notebooks/kkl15.jl

```diff
@@ -0,0 +1,2102 @@
+	@transform!(dat_subj_w,  :object =     :dos - :sod)
+	@transform!(dat_subj_w,  :attraction = :dod - :dos)
+	describe(dat_subj_w)
+end
+
+# ╔═╡ a5226169-dd2e-4105-9569-98fd781332a4
+@df dat_subj_w StatsPlots.corrplot(cols(2:5), grid = false, compact=false)
+
+# ╔═╡ 1e740cc3-3037-4fa3-82d2-75b13b86a593
+@df dat_subj_w StatsPlots.corrplot(cols(6:9), grid = false, compact=false)
+
+# ╔═╡ 7343b5fc-3664-4f13-8a95-8795a1b0c959
+
+
+# ╔═╡ 019a4876-e32d-41f4-9636-7efaea5e4929
+md"**Note:** Two of the theoreticsally irrelevant within-subject effect correlations have a different sign than the corresponding, non-significant CPs in the LMM; they are negative here, numerically positive in the LMM. This occurs only very rarely in the case of ecological correlations. However, as they are not significant according to shortest coverage interval, it may not be that relevant either. It is the case both for effects based on log-transformed and raw reaction times."
+
+# ╔═╡ 317f7e3d-94ef-4a36-a45d-e08b5fe0cd20
+md"""## Linear mixed model
+"""
+
+# ╔═╡ e827e62a-b458-46a7-a86c-b3c3392c3e6a
+begin
+	cntr1 = Dict(
+    :CTR  => SeqDiffCoding(levels=["val", "sod", "dos", "dod"]),
+ 	);
+end
+
+# ╔═╡ bd24f002-3f17-4cea-b371-5a972956da11
+begin	
+	formula = @formula  lrt ~ 1 + CTR + (1 + CTR | Subj);
+	m1 = fit(MixedModel, formula, dat, contrasts=cntr1)
+end
+
+# ╔═╡ 21ab3d31-c385-4520-a8c7-927d8d63f074
+VarCorr(m1)
+
+# ╔═╡ 51cab366-6b2d-4b27-a248-947a0e8032cb
+issingular(m1)
+
+# ╔═╡ 60a3548e-796c-4bdf-95c7-6fefbbb35cb9
+MixedModels.PCA(m1)
+
+# ╔═╡ 960f1f33-d24d-4aa8-a806-e948aeaa711d
+md" We note that the critical correlation parameter between spatial (`sod`) and attraction (`dod`) is now estimated at .66 -- not that close to the 1.0 boundary that caused singularity in Kliegl et al. (2011; `kydwz11.jl`). However, the LMM based on log reaction times is still singular. Let's check for untransformed reaction times."  
+
+# ╔═╡ dbc8c8fb-2ca0-4424-9dfa-458b21c96130
+begin	
+	formula_rt = @formula  rt ~ 1 + CTR + (1 + CTR | Subj);
+	m1_rt = fit(MixedModel, formula_rt, dat, contrasts=cntr1)
+	VarCorr(m1_rt)
+end
+
+# ╔═╡ d46d1bc3-baf5-4e93-8f4f-e3d400e5b0dd
+issingular(m1_rt)
+
+# ╔═╡ 37a7fbf7-68e7-419f-80fc-7c0ad0d30fa3
+md"For untransformed reaction times, we the model is **not** singular."
+
+# ╔═╡ 0ffa0a85-f197-47ce-8e22-4a7667ffd404
+md""" ## Diagnostic plots of LMM residuals
+
+Do model residuals meet LMM assumptions? Classic plots are 
+
++ Residual over fitted
++ Quantiles of model residuals over theoretical quantiles of normal distribution
+
+### Residual-over-fitted plot
+
+The slant in residuals show a lower and upper boundary of reaction times, that is we have have too few short and too few long residuals. Not ideal, but at least width of the residual band looks similar across the fitted values, that is there is no evidence for heteroskedasticity.
+"""
+
+# ╔═╡ e215a6c6-3499-4bd4-bdc5-c153ade0122a
+StatsPlots.scatter(fitted(m1), residuals(m1))
```

Comment:
Do we need to switch to `StatsPlots` here?  I think it can be written as
```
data((; fitted=fitted(m1), resid=residuals(m1)) * mapping(:fitted=>"Fitted values for m1", :resid="Residuals") |> draw
```
(The default `visual` is `Scatter`.)

Reply:
No objection.